### PR TITLE
Memory cache the current logged in member when deciding which environment the member can view

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

File: code/model/DNProject.php [hunk 1]
```diff
@@ -358,9 +363,19 @@ public function getRepository() {
 	 * @return ArrayList
 	 */
 	public function DNEnvironmentList() {
+
+		if(!self::$_current_member_cache) {
+			self::$_current_member_cache = Member::currentUser();
+		}
+
+		if(self::$_current_member_cache === false) {
+			new ArrayList();
```
Comment (halkyon) [inline code/model/DNProject.php]: not assigned to anything here?

**Comment by stojg** [comment]
In my local environment with a lot of environment it decreases the # SQL from 585 to 258, ie removing ~330 duplicate queries

File: code/model/DNProject.php (hunk 1 above)
Comment (stojg) [inline code/model/DNProject.php]: sloppy, Ill fix

Comment (halkyon) [inline code/model/DNProject.php]: The code here is a little confusing. Couldn't you combine the `!` and `false` check into one?

**Comment by stojg** [comment]
In worst case if there is no logged in member it will still do all those queries, but a user shouldn't get this far without a login.

**Comment by halkyon** [comment]
ok nevermind my last comment about confusing. It makes sense now that you return an ArrayList :)

**Pull request merged**